### PR TITLE
tests: filter ubuntu-core systems for authenticated find-private test

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

**Comment by pedronis** [comment]
@fgimenez why doesn't the test work on those systems?

File: tests/main/find-private/task.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 summary: Check that find works with private snaps.
 
 # ppc64el disabled because of https://github.com/snapcore/snapd/issues/2502
-systems: [-ubuntu-16.04-ppc64el, -ubuntu-16.10-ppc64el, -ubuntu-17.04-ppc64el]
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-arm-64, -ubuntu-core-16-arm-32, -ubuntu-16.04-ppc64el, -ubuntu-16.10-ppc64el, -ubuntu-17.04-ppc64el]
```
Comment (mvo5) [inline tests/main/find-private/task.yaml]: (nitpick) with the latest spread you can write: `-ubuntu-core-*` here.

Comment (mvo5) [inline tests/main/find-private/task.yaml]: And a comment why those are disabled would be good

**Comment by fgimenez** [comment]
@pedronis we don't have `expect` available on ubuntu-core systems, the authenticated code path can't be run currently on those systems.

@pedronis @mvo5 i've modified slightly the approach so that we only execute the authenticated checks  on non-ubuntu-core systems, this way the old checks for unauthenticated scenarios will still run on ubuntu-core, please take a look.

**Pull request merged**